### PR TITLE
Add `rounded(numberOfDecimalPlaces:, rule:)` to `BinaryFloatingPoint`

### PR DESCRIPTION
**Pull request by jianstm**
Added `rounded(numberOfDecimalPlaces:, rule:)` to `Double` and `Float` get the rounded value with the specified number of decimal places. eg:

```swift
let num: Double = 3.1415927
num.rounded(numberOfDecimalPlaces: 3, rule: .up) -> 3.142
num.rounded(numberOfDecimalPlaces: 3, rule: .down) -> 3.141
num.rounded(numberOfDecimalPlaces: 2, rule: .awayFromZero) -> 3.15
num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero) -> 3.1415
num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven) -> 3
```

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/SwiftStdlib/DoubleExtensions.swift [hunk 1]
```diff
@@ -30,6 +30,27 @@ public extension Double {
...
+    ///     num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero) -> 3.1415
+    ///     num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven) -> 3
+    ///
+    /// - Parameters:
+    ///   - numberOfDecimalPlaces: The expected number of decimal places.
+    ///   - rule: The rounding rule to use.
+    /// - Returns: The rounded value.
+    func rounded(numberOfDecimalPlaces: Int, rule: FloatingPointRoundingRule) -> Double {
+        let number = numberOfDecimalPlaces < 0 ? 0 : numberOfDecimalPlaces
+        let factor = pow(10.0, Double(number))
```
Comment (guykogus) [inline Sources/Extensions/SwiftStdlib/DoubleExtensions.swift]: You can avoid the line above with
`let factor = pow(10.0, Double(max(0, number)))`

File: Sources/Extensions/SwiftStdlib/FloatExtensions.swift [hunk 2]
```diff
@@ -30,6 +30,27 @@ public extension Float {
...
+    ///     num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero) -> 3.1415
+    ///     num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven) -> 3
+    ///
+    /// - Parameters:
+    ///   - numberOfDecimalPlaces: The expected number of decimal places.
+    ///   - rule: The rounding rule to use.
+    /// - Returns: The rounded value.
+    func rounded(numberOfDecimalPlaces: Int, rule: FloatingPointRoundingRule) -> Float {
+        let number = numberOfDecimalPlaces < 0 ? 0 : numberOfDecimalPlaces
+        let factor = powf(10.0, Float(number))
```
Comment (guykogus) [inline Sources/Extensions/SwiftStdlib/FloatExtensions.swift]: Same simplification using `max` can be done here.

File: Tests/SwiftStdlibTests/DoubleExtensionsTests.swift [hunk 3]
```diff
@@ -27,6 +27,15 @@ final class DoubleExtensionsTests: XCTestCase {
         XCTAssertEqual(Double(4.3).cgFloat, CGFloat(4.3))
     }
 
+    func testRounded() {
+        let num: Double = 3.1415927
+        XCTAssertEqual(num.rounded(numberOfDecimalPlaces: 3, rule: .up), 3.142)
+        XCTAssertEqual(num.rounded(numberOfDecimalPlaces: 3, rule: .down), 3.141)
+        XCTAssertEqual(num.rounded(numberOfDecimalPlaces: 2, rule: .awayFromZero), 3.15)
+        XCTAssertEqual(num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero), 3.1415)
+        XCTAssertEqual(num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven), 3)
```
Comment (guykogus) [inline Tests/SwiftStdlibTests/DoubleExtensionsTests.swift]: Please add test cases for zero as well.

**Comment by jianstm** [comment]
Hi @guykogus , thank you for reviewing! I use this method to calculate weights and measures conversion in my project. We don't want `String` sometimes, and `NumberFormatter` seems very heavy.

File: Sources/Extensions/SwiftStdlib/BinaryFloatingPointExtensions.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    ///     num.rounded(numberOfDecimalPlaces: 3, rule: .down) -> 3.141
+    ///     num.rounded(numberOfDecimalPlaces: 2, rule: .awayFromZero) -> 3.15
+    ///     num.rounded(numberOfDecimalPlaces: 4, rule: .towardZero) -> 3.1415
+    ///     num.rounded(numberOfDecimalPlaces: -1, rule: .toNearestOrEven) -> 3
+    ///
+    /// - Parameters:
+    ///   - numberOfDecimalPlaces: The expected number of decimal places.
+    ///   - rule: The rounding rule to use.
+    /// - Returns: The rounded value.
+    func rounded(numberOfDecimalPlaces: Int, rule: FloatingPointRoundingRule) -> Self {
```
Comment (guykogus) [inline Sources/Extensions/SwiftStdlib/BinaryFloatingPointExtensions.swift]: Please make the function `public`

File: SwifterSwift.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -729,13 +740,13 @@
 				078916D92076077000AC0665 /* FloatingPointExtensionsTests.swift */,
 				07C50D031F5EB03200F46E5A /* IntExtensionsTests.swift */,
 				07C50D051F5EB03200F46E5A /* OptionalExtensionsTests.swift */,
+				B29527B120F9A04900E1F75D /* RandomAccessCollectionExtensionsTests.swift */,
```
Comment (guykogus) [inline SwifterSwift.xcodeproj/project.pbxproj]: Can you please revert the project changes for the files that you didn't change?

File: CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,8 @@ The changelog for **SwifterSwift**. Also see the [releases](https://github.com/S
   - Added `Dictionary[path:]` subscript for deep fetching/setting nested values. [#574](https://github.com/SwifterSwift/SwifterSwift/pull/573) by [@calebkleveter](https://github.com/calebkleveter)
 - **UIColor**
   - Added `whatsApp` color constant. [#581](https://github.com/SwifterSwift/SwifterSwift/pull/581) by [staffler-xyz](https://github.com/staffler-xyz)
+- **BinaryFloatingPointExtensions**
+  - Added `rounded(numberOfDecimalPlaces:, rule:)` to get the rounded floating number with the specified number of decimal places. [#583](https://github.com/SwifterSwift/SwifterSwift/pull/583) by [jianstm](https://github.com/jianstm)
```
Comment (guykogus) [inline CHANGELOG.md]: Minor note: The function signature is `rounded(numberOfDecimalPlaces:rule:)`

Comment (jianstm) [reply CHANGELOG.md]: Fixed! Very thank you!

**Comment by SD10** [comment]
Thank you for contributing to SwifterSwift! I've invited you to join the SwifterSwift GitHub organization - no pressure to accept! If you'd like more information on what that means, check out our [contributing guidelines](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md). Feel free to reach out if you have any questions! 😃
